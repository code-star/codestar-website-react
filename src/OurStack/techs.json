{
	"backEnd": [
		{
			"logo": "/images/technologies/scala.svg",
			"name": "Scala"
		},
		{
			"logo": "/images/technologies/akka.svg",
			"name": "Akka"
		},
		{
			"logo": "/images/technologies/spray.png",
			"name": "Spray"
		},
		{
			"logo": "/images/technologies/spark.png",
			"name": "Spark"
		},
		{
			"logo": "/images/technologies/cassandra.png",
			"name": "Cassandra"
		},
		{
			"logo": "/images/technologies/kafka.png",
			"name": "Kafka"
		}
	],
	"frontEnd": [
		{
			"logo": "/images/technologies/typescript.png",
			"name": "Typescript"
		},
		{
			"logo": "/images/technologies/react.png",
			"name": "React"
		},
		{
			"logo": "/images/technologies/redux.png",
			"name": "Redux"
		},
		{
			"logo": "/images/technologies/angular.png",
			"name": "Angular"
		},
		{
			"logo": "/images/technologies/rxjs.png",
			"name": "RxJS"
		}
	],
	"infrastructure": [
		{
			"logo": "/images/technologies/docker.png",
			"name": "Docker"
		},
		{
			"logo": "/images/technologies/mesos.jpg",
			"name": "Mesos"
		},
		{
			"logo": "/images/technologies/kubernetes.png",
			"name": "Kubernetes"
		},
		{
			"logo": "/images/technologies/aws.png",
			"name": "AWS"
		}
	]
}
