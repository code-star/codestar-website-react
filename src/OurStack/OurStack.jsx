import React, { Component } from 'react';
import { translate } from 'react-i18next';

import { Paper, Avatar, Tooltip, Typography, Grow } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';

import { responsiveImageProps } from '../ResponsiveImage/ResponsiveImage';
import techs from './techs';

const styles = theme => ({
	root: {
		...theme.mixins.gutters(),
		marginBottom: theme.spacing.unit * 1,
		paddingTop: theme.spacing.unit * 1.5,
		paddingBottom: theme.spacing.unit * 1.5,
	},
	icon: {
		marginLeft: 10,
	},
});

@translate(['stack'], { wait: true })
class OurStack extends Component {
	render() {
		const { t, classes } = this.props;

		function techIcons(techs) {
			return (
				<div className="row">
					{techs.map(item => (
						<Tooltip
							key={item.logo}
							title={item.name} // TODO: Description why we chose a tech
						>
							<Grow in>
								<Avatar
									className={classes.icon}
									{...responsiveImageProps(item.logo, item.name)}
								/>
							</Grow>
						</Tooltip>
					))}
				</div>
			);
		}

		const parts = [
			{
				title: 'STACK_FRONT_END',
				caption: 'STACK_FRONT_END_SUBTITLE',
				icons: techs.frontEnd,
			},
			{
				title: 'STACK_BACK_END',
				caption: 'STACK_BACK_END_SUBTITLE',
				icons: techs.backEnd,
			},
			{
				title: 'STACK_INFRASTRUCTURE',
				caption: 'STACK_INFRASTRUCTURE_SUBTITLE',
				icons: techs.infrastructure,
			},
		];

		return (
			<div>
				{/* TODO: better handling of small screen size */}
				{parts.map(part => (
					<Paper key={part.title} className={classes.root} elevation={1}>
						<div className="row justify-content-between">
							<div className="col-auto mb-1">
								{t(part.title)}
								<Typography variant="caption">{t(part.caption)}</Typography>
							</div>
							<div className="col-auto mr-1">{techIcons(part.icons)}</div>
						</div>
					</Paper>
				))}
			</div>
		);
	}
}

export default withStyles(styles)(OurStack);
