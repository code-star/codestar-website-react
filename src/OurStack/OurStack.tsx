import * as React from "react";
import { translate } from "react-i18next";

import { Paper, Avatar, Tooltip, Typography, Grow } from "@material-ui/core";
import { withStyles } from "@material-ui/core/styles";

import { getResponsiveImageUrl } from "../ResponsiveImage/ResponsiveImage";
import techs from "./techs.json";

type OurStackProps = any;

const styles = (theme: any) => ({
	root: {
		...theme.mixins.gutters(),
		marginBottom: theme.spacing.unit * 1,
		paddingTop: theme.spacing.unit * 1.5,
		paddingBottom: theme.spacing.unit * 1.5
	},
	icon: {
		marginLeft: 10,
		width: 40,
		height: 40
	}
});

class OurStack extends React.Component<OurStackProps> {
	public render() {
		const { t, classes } = this.props;

		function techIcons(techItems: any) {
			return (
				<div className="row">
					{techItems.map((item: any) => (
						<Tooltip
							key={item.logo}
							title={item.name} // TODO: Description why we chose a tech
						>
							<Grow in={true}>
								<Avatar
									className={classes.icon}
									alt={item.name}
									src={getResponsiveImageUrl(item.logo, 80)}
								/>
							</Grow>
						</Tooltip>
					))}
				</div>
			);
		}

		const parts = [
			{
				title: "STACK_FRONT_END",
				caption: "STACK_FRONT_END_SUBTITLE",
				icons: techs.frontEnd
			},
			{
				title: "STACK_BACK_END",
				caption: "STACK_BACK_END_SUBTITLE",
				icons: techs.backEnd
			},
			{
				title: "STACK_INFRASTRUCTURE",
				caption: "STACK_INFRASTRUCTURE_SUBTITLE",
				icons: techs.infrastructure
			}
		];

		return (
			<div>
				{/* TODO: better handling of small screen size */}
				{parts.map(part => (
					<Paper key={part.title} className={classes.root} elevation={1}>
						<div className="row justify-content-between">
							<div className="col-auto mb-1">
								{t(part.title)}
								<Typography variant="caption">{t(part.caption)}</Typography>
							</div>
							<div className="col-auto mr-1">{techIcons(part.icons)}</div>
						</div>
					</Paper>
				))}
			</div>
		);
	}
}

export default translate(["stack"], { wait: true })(
	withStyles(styles)(OurStack)
);
