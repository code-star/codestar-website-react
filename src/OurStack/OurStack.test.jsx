import React from 'react';
import { OurStack } from './OurStack';
import renderer from 'react-test-renderer';

jest.mock('@material-ui/core/Grow', () => Component => (
  <div>{Component.children}</div>
));

describe('OurStack', () => {
  it('shows icons for each technology', () => {
    const comp = renderer
      .create(<OurStack classes={{ root: '' }} t={() => ''} />)
      .toJSON();
    expect(comp).toMatchSnapshot();
  });
});
