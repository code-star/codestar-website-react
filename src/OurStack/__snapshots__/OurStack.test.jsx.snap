// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OurStack shows icons for each technology 1`] = `
<div>
  <div
    className="MuiPaper-root-1 MuiPaper-elevation1-4 MuiPaper-rounded-2"
  >
    <div
      className="row justify-content-between"
    >
      <div
        className="col-auto mb-1"
      >
        
        <span
          className="MuiTypography-root-28 MuiTypography-caption-38"
        >
          
        </span>
      </div>
      <div
        className="col-auto mr-1"
      >
        <div
          className="row"
        >
          <div>
            <div
              className="MuiAvatar-root-70"
            >
              <img
                alt="Typescript"
                className="MuiAvatar-img-72"
                src="https://res.cloudinary.com/codestar/image/upload/f_auto/w_80/v1542784068/codestar.nl/images/technologies/typescript.png"
              />
            </div>
          </div>
          <div>
            <div
              className="MuiAvatar-root-70"
            >
              <img
                alt="React"
                className="MuiAvatar-img-72"
                src="https://res.cloudinary.com/codestar/image/upload/f_auto/w_80/v1542784068/codestar.nl/images/technologies/react.png"
              />
            </div>
          </div>
          <div>
            <div
              className="MuiAvatar-root-70"
            >
              <img
                alt="Redux"
                className="MuiAvatar-img-72"
                src="https://res.cloudinary.com/codestar/image/upload/f_auto/w_80/v1542784068/codestar.nl/images/technologies/redux.png"
              />
            </div>
          </div>
          <div>
            <div
              className="MuiAvatar-root-70"
            >
              <img
                alt="Angular"
                className="MuiAvatar-img-72"
                src="https://res.cloudinary.com/codestar/image/upload/f_auto/w_80/v1542784068/codestar.nl/images/technologies/angular.png"
              />
            </div>
          </div>
          <div>
            <div
              className="MuiAvatar-root-70"
            >
              <img
                alt="RxJS"
                className="MuiAvatar-img-72"
                src="https://res.cloudinary.com/codestar/image/upload/f_auto/w_80/v1542784068/codestar.nl/images/technologies/rxjs.png"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root-1 MuiPaper-elevation1-4 MuiPaper-rounded-2"
  >
    <div
      className="row justify-content-between"
    >
      <div
        className="col-auto mb-1"
      >
        
        <span
          className="MuiTypography-root-28 MuiTypography-caption-38"
        >
          
        </span>
      </div>
      <div
        className="col-auto mr-1"
      >
        <div
          className="row"
        >
          <div>
            <div
              className="MuiAvatar-root-70"
            >
              <img
                alt="Scala"
                className="MuiAvatar-img-72"
                src="https://res.cloudinary.com/codestar/image/upload/f_auto/w_80/v1542784068/codestar.nl/images/technologies/scala.svg"
              />
            </div>
          </div>
          <div>
            <div
              className="MuiAvatar-root-70"
            >
              <img
                alt="Akka"
                className="MuiAvatar-img-72"
                src="https://res.cloudinary.com/codestar/image/upload/f_auto/w_80/v1542784068/codestar.nl/images/technologies/akka.svg"
              />
            </div>
          </div>
          <div>
            <div
              className="MuiAvatar-root-70"
            >
              <img
                alt="Spray"
                className="MuiAvatar-img-72"
                src="https://res.cloudinary.com/codestar/image/upload/f_auto/w_80/v1542784068/codestar.nl/images/technologies/spray.png"
              />
            </div>
          </div>
          <div>
            <div
              className="MuiAvatar-root-70"
            >
              <img
                alt="Spark"
                className="MuiAvatar-img-72"
                src="https://res.cloudinary.com/codestar/image/upload/f_auto/w_80/v1542784068/codestar.nl/images/technologies/spark.png"
              />
            </div>
          </div>
          <div>
            <div
              className="MuiAvatar-root-70"
            >
              <img
                alt="Cassandra"
                className="MuiAvatar-img-72"
                src="https://res.cloudinary.com/codestar/image/upload/f_auto/w_80/v1542784068/codestar.nl/images/technologies/cassandra.png"
              />
            </div>
          </div>
          <div>
            <div
              className="MuiAvatar-root-70"
            >
              <img
                alt="Kafka"
                className="MuiAvatar-img-72"
                src="https://res.cloudinary.com/codestar/image/upload/f_auto/w_80/v1542784068/codestar.nl/images/technologies/kafka.png"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root-1 MuiPaper-elevation1-4 MuiPaper-rounded-2"
  >
    <div
      className="row justify-content-between"
    >
      <div
        className="col-auto mb-1"
      >
        
        <span
          className="MuiTypography-root-28 MuiTypography-caption-38"
        >
          
        </span>
      </div>
      <div
        className="col-auto mr-1"
      >
        <div
          className="row"
        >
          <div>
            <div
              className="MuiAvatar-root-70"
            >
              <img
                alt="Docker"
                className="MuiAvatar-img-72"
                src="https://res.cloudinary.com/codestar/image/upload/f_auto/w_80/v1542784068/codestar.nl/images/technologies/docker.png"
              />
            </div>
          </div>
          <div>
            <div
              className="MuiAvatar-root-70"
            >
              <img
                alt="Mesos"
                className="MuiAvatar-img-72"
                src="https://res.cloudinary.com/codestar/image/upload/f_auto/w_80/v1542784068/codestar.nl/images/technologies/mesos.jpg"
              />
            </div>
          </div>
          <div>
            <div
              className="MuiAvatar-root-70"
            >
              <img
                alt="Kubernetes"
                className="MuiAvatar-img-72"
                src="https://res.cloudinary.com/codestar/image/upload/f_auto/w_80/v1542784068/codestar.nl/images/technologies/kubernetes.png"
              />
            </div>
          </div>
          <div>
            <div
              className="MuiAvatar-root-70"
            >
              <img
                alt="AWS"
                className="MuiAvatar-img-72"
                src="https://res.cloudinary.com/codestar/image/upload/f_auto/w_80/v1542784068/codestar.nl/images/technologies/aws.png"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
